tests/unit_test/twitter_ads_api/twitter_ads_api_pipeline_test.py

```diff
@@ -267,11 +338,139 @@ def test_should_pass_correct_values_to_get_param_dict_func_if_api_query_paramete
+                placement_value=['placement_value_1', 'placement_value_2']
+            )
+        )
+        list(iter_bq_compatible_json_response_from_resource_with_provenance(
+            SOURCE_CONFIG_WITH_API_QUERY_PARAMETERS_IN_PLACEMENT._replace(
+                api_query_parameters=api_query_parameters
+            )
+        ))
+
+        get_param_dict_from_api_query_parameters_mock.assert_has_calls(calls=[
+            call(
+                api_query_parameters_config=api_query_parameters,
+                value_from_bq='id_1',
+                start_time='2022-08-01',
+                end_time='2022-08-02',
+                placement='placement_value_1'
+            ),
+            call(
+                api_query_parameters_config=api_query_parameters,
+                value_from_bq='id_1',
+                start_time='2022-08-01',
+                end_time='2022-08-02',
+                placement='placement_value_2'
+            )
+        ], any_order=True)
+
+    def test_should_call_get_param_dict_for_every_seven_days_from_start_date_till_yesterday(
+        self,
+        iter_dict_from_bq_query_for_bigquery_source_config_mock: MagicMock,
+        get_yesterdays_date_mock: MagicMock,
+        get_param_dict_from_api_query_parameters_mock: MagicMock
+    ):
+        iter_dict_from_bq_query_for_bigquery_source_config_mock.return_value = (
+            [{'campaign_id': 'id_1', 'create_date': '2022-08-01'}]
+        )
+        get_yesterdays_date_mock.return_value = datetime(2022, 8, 17).date()
+        api_query_parameters = API_QUERY_PARAMETERS_WITH_PLACEMENT._replace(
+            parameter_values=PARAMETER_VALUES_WITH_PLACEMENT._replace(
+                start_time_value=None,
+                end_time_value=None
+            )
+        )
+        list(iter_bq_compatible_json_response_from_resource_with_provenance(
+            SOURCE_CONFIG_WITH_API_QUERY_PARAMETERS_IN_PLACEMENT._replace(
+                api_query_parameters=api_query_parameters
+            )
+        ))
+        # print('call_list: ', get_param_dict_from_api_query_parameters_mock.call_args_list)
+
+        get_param_dict_from_api_query_parameters_mock.assert_has_calls(calls=[
+            call(
+                api_query_parameters_config=api_query_parameters,
+                value_from_bq='id_1',
+                start_time='2022-08-01',
+                end_time='2022-08-08',
+                placement=PLACEMENT_PARAM_VALUE[0]
+            ),
+            call(
+                api_query_parameters_config=api_query_parameters,
+                value_from_bq='id_1',
+                start_time='2022-08-08',
+                end_time='2022-08-15',
+                placement=PLACEMENT_PARAM_VALUE[0]
+            ),
+            call(
+                api_query_parameters_config=api_query_parameters,
+                value_from_bq='id_1',
+                start_time='2022-08-15',
+                end_time='2022-08-17',
```

Comment:
mayeb add a comment regarding this being yesterday as +7 days would be in the future?